codex-rs/mcp-server/src/mcp_protocol.rs

```diff
@@ -172,15 +172,21 @@ pub enum ToolCallResponseResult {
 }
 
 #[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
-pub struct ConversationCreateResult {
-    pub conversation_id: ConversationId,
-    pub model: String,
+pub enum ConversationCreateResult {
+    Ok {
+        conversation_id: ConversationId,
+        model: String,
+    },
+    Error {
```

Comment:
Does this needs its own error type or can this be returned as a McpRequest result with an error?